projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -0,0 +1,531 @@
+  ]
+})
+export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnInit, ControlValueAccessor {
+  /**
+  * An @Input property that allows you to set the locale settings used in `displayFormat`.
+  * @example
+  *```html
+  * <input igxDateTimeEditor [locale]="'en'">
+  *```
+  */
+  @Input()
+  public locale = 'en';
+
+  /**
+    * An @Input property that allows you to set the minimum possible value the editor will allow.
+    * @example
+    *```html
+    * <input igxDateTimeEditor [minValue]="minDate">
+    *```
+    */
+  @Input()
+  public minValue: string | Date;
+
+  /**
+  * An @Input property that allows you to set the maximum possible value the editor will allow.
+  * @example
+  *```html
+  * <input igxDateTimeEditor [maxValue]="maxDate">
+  *```
+  */
+  @Input()
+  public maxValue: string | Date;
+
+  /**
+   * An @Input property that allows you to specify if the currently spun date segment should loop over.
+   * @example
+   *```html
+   * <input igxDateTimeEditor [isSpinLoop]="false">
+   *```
+   */
+  @Input()
+  public isSpinLoop = true;
+
+  /**
+   * An @Input property that allows you to set both pre-defined format options such as `shortDate` and `longDate`,
+   * as well as constructed format string using characters supported by `DatePipe`, e.g. `EE/MM/yyyy`.
+   * @example
+   *```html
+   * <input igxDateTimeEditor [displayFormat]="'shortDate'">
+   *```
+   */
+  @Input()
+  public displayFormat: string;
+
+  /**
+   * An @Input property that allows you to get/set the expected user input format(and placeholder).
+   *  for the editor.
+   * @example
+   *```html
+   * <input [igxDateTimeEditor]="'dd/MM/yyyy'">
+   *```
+   */
+  @Input(`igxDateTimeEditor`)
+  public set inputFormat(value: string) {
+    if (value) {
+      this._format = value;
+    }
+    const mask = this.buildMask(this.inputFormat);
+    this.mask = value.indexOf('tt') !== -1 ? mask.substring(0, mask.length - 2) + 'LL' : mask;
+  }
+
+  public get inputFormat(): string {
+    return this._format;
+  }
+
+  /**
+   * An @Input property that gets/sets the component date value.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor [value]="date">
+   * ```
+   */
+  @Input()
+  public set value(value: Date) {
+    this._value = value;
+    this.updateMask();
+  }
+
+  public get value() {
+    return this._value;
+  }
+
+  /**
+   * Emitted when the editor's value has changed.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor (valueChanged)="onValueChanged($event)"/>
+   * ```
+   */
+  @Output()
+  public valueChanged = new EventEmitter<IgxDateTimeEditorEventArgs>();
+
+  /**
+   * Emitted when the editor is not within a specified range.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor [minValue]="minDate" [maxValue]="maxDate" (validationFailed)="onValidationFailed($event)"/>
+   * ```
+   */
+  @Output()
+  public validationFailed = new EventEmitter<IgxDateTimeEditorEventArgs>();
```

Comment:
Kinda odd this emits the same arguments - if the validation failed the new value should either clear or revert, but since this will emit _only_ when the user enters something that's not a valid date - that input can't be provided as `Date`. It might be helpful to provide it as a separate arg. And if we do emit the new value, perhaps also respect changes to it from the event handler to allow the app to handle either reset or revert invalid behavior.